File: src/ReadStorm.Desktop/ViewModels/DiagnosticViewModel.cs
using System;
using System.Collections.ObjectModel;
using System.Linq;
using System.Threading;
using System.Threading.Tasks;
using CommunityToolkit.Mvvm.ComponentModel;
using CommunityToolkit.Mvvm.Input;
using ReadStorm.Application.Abstractions;
using ReadStorm.Domain.Models;

namespace ReadStorm.Desktop.ViewModels;

public sealed partial class DiagnosticViewModel : ViewModelBase
{
    private readonly MainWindowViewModel _parent;
    private readonly ISourceDiagnosticUseCase _sourceDiagnosticUseCase;
    private readonly Dictionary<int, SourceDiagnosticResult> _diagnosticResults = new();

    public DiagnosticViewModel(MainWindowViewModel parent, ISourceDiagnosticUseCase sourceDiagnosticUseCase)
    {
        _parent = parent;
        _sourceDiagnosticUseCase = sourceDiagnosticUseCase;
    }

    [ObservableProperty]
    private bool isDiagnosing;

    [ObservableProperty]
    private string diagnosticSummary = string.Empty;

    [ObservableProperty]
    private string? selectedDiagnosticSource;

    partial void OnSelectedDiagnosticSourceChanged(string? value)
    {
        DiagnosticLines.Clear();
        if (value is null) return;

        var match = System.Text.RegularExpressions.Regex.Match(value, @"\[(\d+)\]");
        if (match.Success && int.TryParse(match.Groups[1].Value, out var id)
            && _diagnosticResults.TryGetValue(id, out var result))
        {
            var header = $"[{result.SourceName}] {result.Summary} | HTTP={result.HttpStatusCode} | " +
                         $"搜索={result.SearchResultCount}条 | 目录selector='{result.TocSelector}' " +
                         $"| 章节selector='{result.ChapterContentSelector}'";
            DiagnosticLines.Add(header);
            DiagnosticLines.Add(new string('─', 60));
            foreach (var line in result.DiagnosticLines)
                DiagnosticLines.Add(line);
        }
    }

    public ObservableCollection<string> DiagnosticSourceNames { get; } = [];
    public ObservableCollection<string> DiagnosticLines { get; } = [];

    [RelayCommand]
    private async Task RunBatchDiagnosticAsync()
    {
        try
        {
            IsDiagnosing = true;
            DiagnosticSummary = "正在批量诊断所有书源…";
            DiagnosticLines.Clear();
            _diagnosticResults.Clear();
            DiagnosticSourceNames.Clear();

            var rules = _parent.Sources.Where(s => s.Id > 0).ToList();
            var total = rules.Count;
            var completed = 0;
            var healthy = 0;

            var tasks = rules.Select(async source =>
            {
                var result = await _sourceDiagnosticUseCase.DiagnoseAsync(source.Id, "测试");
                Interlocked.Increment(ref completed);
                if (result.IsHealthy) Interlocked.Increment(ref healthy);
                return result;
            });

            var results = await Task.WhenAll(tasks);
            foreach (var r in results.OrderBy(r => r.SourceId))
            {
                _diagnosticResults[r.SourceId] = r;
                var prefix = r.IsHealthy ? "🟢" : "🔴";
                DiagnosticSourceNames.Add($"{prefix} [{r.SourceId}] {r.SourceName}");
            }

            DiagnosticSummary = $"批量诊断完成：{healthy}/{total} 个书源正常";
            _parent.StatusMessage = DiagnosticSummary;
            if (DiagnosticSourceNames.Count > 0)
                SelectedDiagnosticSource = DiagnosticSourceNames[0];
        }
        catch (Exception ex)
        {
            DiagnosticSummary = $"诊断异常：{ex.Message}";
            _parent.StatusMessage = $"诊断失败：{ex.Message}";
        }
        finally { IsDiagnosing = false; }
    }
}

File: pureopus/app/src/main/java/com/readstorm/app/ui/viewmodels/DiagnosticViewModel.kt
package com.readstorm.app.ui.viewmodels

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData

class DiagnosticViewModel(
    private val parent: MainViewModel
) {
    private val _isDiagnosing = MutableLiveData(false)
    val isDiagnosing: LiveData<Boolean> = _isDiagnosing

    private val _diagnosticSummary = MutableLiveData("")
    val diagnosticSummary: LiveData<String> = _diagnosticSummary

    private val _selectedDiagnosticSource = MutableLiveData<String?>(null)
    val selectedDiagnosticSource: LiveData<String?> = _selectedDiagnosticSource

    private val _diagnosticSourceNames = MutableLiveData<List<String>>(emptyList())
    val diagnosticSourceNames: LiveData<List<String>> = _diagnosticSourceNames

    private val _diagnosticLines = MutableLiveData<List<String>>(emptyList())
    val diagnosticLines: LiveData<List<String>> = _diagnosticLines

    private val diagnosticResults = mutableMapOf<Int, com.readstorm.app.domain.models.SourceDiagnosticResult>()

    fun setSelectedDiagnosticSource(source: String?) {
        _selectedDiagnosticSource.postValue(source)
        if (source == null) {
            _diagnosticLines.postValue(emptyList())
            return
        }

        val match = Regex("\\[(\\d+)\\]").find(source)
        if (match != null) {
            val id = match.groupValues[1].toIntOrNull()
            if (id != null && diagnosticResults.containsKey(id)) {
                val result = diagnosticResults[id]!!
                val lines = mutableListOf<String>()
                lines.add("[${result.sourceName}] ${result.summary} | HTTP=${result.httpStatusCode} | " +
                    "搜索=${result.searchResultCount}条 | 目录selector='${result.tocSelector}' " +
                    "| 章节selector='${result.chapterContentSelector}'")
                lines.add("─".repeat(60))
                lines.addAll(result.diagnosticLines)
                _diagnosticLines.postValue(lines)
            }
        }
    }

    suspend fun runBatchDiagnostic() {
        try {
            _isDiagnosing.postValue(true)
            _diagnosticSummary.postValue("正在批量诊断所有书源…")
            _diagnosticLines.postValue(emptyList())
            diagnosticResults.clear()
            _diagnosticSourceNames.postValue(emptyList())

            val rules = parent.sources.filter { it.id > 0 }
            val total = rules.size
            var healthyCount = 0
            val sourceNames = mutableListOf<String>()

            for ((index, source) in rules.withIndex()) {
                _diagnosticSummary.postValue("正在诊断 (${index + 1}/$total): ${source.name}…")
                try {
                    val result = parent.diagnosticUseCase.diagnose(source.id, "测试")
                    diagnosticResults[source.id] = result
                    val emoji = if (result.isHealthy) "🟢" else "🔴"
                    sourceNames.add("$emoji [${source.id}] ${source.name}")
                    if (result.isHealthy) healthyCount++
                } catch (_: Exception) {
                    sourceNames.add("🔴 [${source.id}] ${source.name}")
                }
            }

            _diagnosticSourceNames.postValue(sourceNames)
            _diagnosticSummary.postValue("诊断完成：$healthyCount/$total 个书源健康")
            parent.setStatusMessage("批量诊断完成：$healthyCount/$total 个书源健康")
        } catch (e: Exception) {
            _diagnosticSummary.postValue("诊断异常：${e.message}")
            parent.setStatusMessage("诊断失败：${e.message}")
        } finally {
            _isDiagnosing.postValue(false)
        }
    }
}
